src/commands/watch.rs

```diff
@@ -185,47 +180,63 @@ impl WatchCommand {
             .await?
             .into_v1();
 
-        let path_patterns: Vec<String> = match self.watch_assets {
-            // Watch patterns
-            false => app_manifest
-                .components
-                .iter()
-                .filter_map(|c| c.build.as_ref())
-                .filter_map(|b| b.watch.clone())
-                .flatten()
-                .collect(),
-            // Asset patterns
-            true => {
-                let component_source_patterns = app_manifest
-                    .components
-                    .iter()
-                    .filter_map(|c| {
-                        if let RawModuleSource::FileReference(path) = &c.source {
-                            return path.to_str();
+        let watch_patterns = app_manifest
+            .components
+            .iter()
+            .filter_map(|c| {
+                if let Some(build) = c.build.as_ref() {
+                    if build.watch.is_none() {
+                        // TODO: Make sure that documentation link is pointing specifically to spin watch docs
+                        println!("You haven't configured what to watch for the component: '{}'. Learn how to configure Spin watch at https://developer.fermyon.com",c.id);
+                    }
+                    build.watch.clone()
+                } else {
+                    // No build config for this component so lets watch the source instead
+                    if let RawModuleSource::FileReference(path) = &c.source {
+                        if let Some(string_path) = path.to_str() {
+                            return Some(vec![String::from(string_path)]);
                         }
-                        None
-                    })
-                    .map(String::from);
-                let component_file_patterns = app_manifest
-                    .components
-                    .iter()
-                    .filter_map(|c| c.wasm.files.as_ref())
-                    .flatten()
-                    .map(|raw_file_mount| match raw_file_mount {
-                        RawFileMount::Placement(raw_directory_placement) => String::from(
-                            raw_directory_placement
-                                .source
-                                .join("**/*")
-                                .to_str()
-                                .expect("conversion to str not to fail"),
-                        ),
-                        RawFileMount::Pattern(pattern) => pattern.to_string(),
-                    });
-                component_source_patterns
-                    .chain(component_file_patterns)
-                    .collect()
-            }
+                    }
+                    None
+                }
+            })
+            .flatten();
+        let component_source_patterns = app_manifest
+            .components
+            .iter()
+            .filter_map(|c| {
```

Comment:
This can be deferred to a later refactoring pass but it bothers me that we have two paths that lead to watching the `component.source` in slightly different ways (but one of them gets thrown away depending on the `--skip-build` flag).  I think we can tweak this to be a bit neater, but we can do that over time, not a blocker for now.

Reply:
I agree. This function was a mess and I didn't know how to clean it up. I feel like I've been too close to this code for too long. We'll clean this up in the future. I'll drop a TODO.